universal-application-tool-0.0.1/app/services/question/MultiOptionQuestionDefinition.java

```diff
@@ -0,0 +1,109 @@
+package services.question;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.OptionalLong;
+import services.Path;
+
+public abstract class MultiOptionQuestionDefinition extends QuestionDefinition {
+
+  private final ImmutableListMultimap<Locale, String> options;
+
+  protected MultiOptionQuestionDefinition(
+      OptionalLong id,
+      long version,
+      String name,
+      Path path,
+      String description,
+      ImmutableMap<Locale, String> questionText,
+      ImmutableMap<Locale, String> questionHelpText,
+      ImmutableListMultimap<Locale, String> options) {
+    super(
+        id,
+        version,
+        name,
+        path,
+        description,
+        questionText,
+        questionHelpText,
+        MultiOptionValidationPredicates.create());
```

Comment:
Thoughts (non-blocking): Do we need specific validation predicates for each concrete multi-option question types? I presume it would be necessary but I am not 100% sure.

Reply:
Great question! For now, the only [planned validation logic](https://docs.google.com/document/d/1HYVHBno6dk7fmS5w6V6Y3jxERHGfDoQKTtzBZr27Y6o/edit#heading=h.1i512171vbiq) across all four types is "min choices required" and "max choices allowed" - this is for multi-select questions like checkbox. For others, the default validation will be to set each of these to 1, so only 1 selection is valid